### PR TITLE
Hubs/Scopes Merge 12 - Replace `IHub` with `IScopes` in Spring 6 / Spring Boot 3 integrations

### PR DESCRIPTION
**Pull request by adinauer**
#skip-changelog

## :scroll: Description

Follow up for https://github.com/getsentry/sentry-java/pull/3297 changing Spring 6 / Spring Boot 3 integrations.

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-spring-boot-jakarta/src/main/java/io/sentry/spring/boot/jakarta/SentryAutoConfiguration.java
Comment (lbloder) [inline sentry-spring-boot-jakarta/src/main/java/io/sentry/spring/boot/jakarta/SentryAutoConfiguration.java:1]: Should we rename `HubConfiguration` in `SentryAutoConfiguration`?

File: sentry-spring-boot-jakarta/src/main/java/io/sentry/spring/boot/jakarta/SentryAutoConfiguration.java [hunk 1]
```diff
@@ -116,7 +116,7 @@ static class HubConfiguration {
     }
 
     @Bean
-    public @NotNull IHub sentryHub(
+    public @NotNull IScopes sentryHub(
```
Comment (lbloder) [inline sentry-spring-boot-jakarta/src/main/java/io/sentry/spring/boot/jakarta/SentryAutoConfiguration.java:119]: Should we rename this method?

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/AbstractSentryWebFilter.java [hunk 2]
```diff
@@ -34,16 +34,17 @@
...
-    Objects.requireNonNull(hub, "hub is required");
-    this.sentryRequestResolver = new SentryRequestResolver(hub);
+  public AbstractSentryWebFilter(final @NotNull IScopes scopes) {
+    Objects.requireNonNull(scopes, "scopes are required");
+    this.sentryRequestResolver = new SentryRequestResolver(scopes);
   }
 
   protected @Nullable ITransaction maybeStartTransaction(
-      final @NotNull IHub requestHub, final @NotNull ServerHttpRequest request) {
+      final @NotNull IScopes requestHub, final @NotNull ServerHttpRequest request) {
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/AbstractSentryWebFilter.java:47]: Should we rename this to something like `requestScopes`?

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/AbstractSentryWebFilter.java [hunk 3]
```diff
@@ -62,21 +63,23 @@ public AbstractSentryWebFilter(final @NotNull IHub hub) {
 
   protected void doFinally(
       final @NotNull ServerWebExchange serverWebExchange,
-      final @NotNull IHub requestHub,
+      final @NotNull IScopes requestHub,
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/AbstractSentryWebFilter.java:66]: Should we rename this to something like `requestScopes`?

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/AbstractSentryWebFilter.java [hunk 4]
```diff
@@ -62,21 +63,23 @@ public AbstractSentryWebFilter(final @NotNull IHub hub) {
...
+      // TODO close lifecycle token instead of popscope
       requestHub.popScope();
     }
-    Sentry.setCurrentHub(NoOpHub.getInstance());
+    Sentry.setCurrentScopes(NoOpScopes.getInstance());
   }
 
   protected void doFirst(
-      final @NotNull ServerWebExchange serverWebExchange, final @NotNull IHub requestHub) {
+      final @NotNull ServerWebExchange serverWebExchange, final @NotNull IScopes requestHub) {
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/AbstractSentryWebFilter.java:79]: Should we rename this to something like `requestScopes`?

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java [hunk 5]
```diff
@@ -1,33 +1,36 @@
...
    * <p>This requires - reactor.core.publisher.Hooks#enableAutomaticContextPropagation() to be
    * enabled - having `io.micrometer:context-propagation:1.0.2+` (provided by Spring Boot 3.0.3+) -
    * having `io.projectreactor:reactor-core:3.5.3+` (provided by Spring Boot 3.0.3+)
    */
   @ApiStatus.Experimental
+  @SuppressWarnings("deprecation")
   public static <T> Mono<T> withSentry(final @NotNull Mono<T> mono) {
-    final @NotNull IHub oldHub = Sentry.getCurrentHub();
-    final @NotNull IHub clonedHub = oldHub.clone();
+    final @NotNull IScopes oldHub = Sentry.getCurrentScopes();
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java:26]: Rename to `oldScopes` and `clonedScopes`/`forkedScopes`?

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java [hunk 6]
```diff
@@ -36,20 +39,20 @@ public static <T> Mono<T> withSentry(final @NotNull Mono<T> mono) {
    */
   @ApiStatus.Experimental
   public static <T> Mono<T> withSentryNewMainHubClone(final @NotNull Mono<T> mono) {
-    final @NotNull IHub hub = Sentry.cloneMainHub();
+    final @NotNull IScopes hub = Sentry.cloneMainHub();
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java:42]: Rename to `scopes`

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java [hunk 7]
```diff
@@ -36,20 +39,20 @@ public static <T> Mono<T> withSentry(final @NotNull Mono<T> mono) {
...
+   * Writes the given Sentry {@link IScopes} to the {@link Context} and uses {@link
    * io.micrometer.context.ThreadLocalAccessor} to propagate it.
    *
    * <p>This requires - reactor.core.publisher.Hooks#enableAutomaticContextPropagation() to be
    * enabled - having `io.micrometer:context-propagation:1.0.2+` (provided by Spring Boot 3.0.3+) -
    * having `io.projectreactor:reactor-core:3.5.3+` (provided by Spring Boot 3.0.3+)
    */
   @ApiStatus.Experimental
-  public static <T> Mono<T> withSentryHub(final @NotNull Mono<T> mono, final @NotNull IHub hub) {
+  public static <T> Mono<T> withSentryHub(final @NotNull Mono<T> mono, final @NotNull IScopes hub) {
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java:55]: Rename to `scopes`

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java [hunk 8]
```diff
@@ -62,23 +65,25 @@ public static <T> Mono<T> withSentryHub(final @NotNull Mono<T> mono, final @NotN
...
    * <p>This requires - reactor.core.publisher.Hooks#enableAutomaticContextPropagation() to be
    * enabled - having `io.micrometer:context-propagation:1.0.2+` (provided by Spring Boot 3.0.3+) -
    * having `io.projectreactor:reactor-core:3.5.3+` (provided by Spring Boot 3.0.3+)
    */
   @ApiStatus.Experimental
+  @SuppressWarnings("deprecation")
   public static <T> Flux<T> withSentry(final @NotNull Flux<T> flux) {
-    final @NotNull IHub oldHub = Sentry.getCurrentHub();
-    final @NotNull IHub clonedHub = oldHub.clone();
+    final @NotNull IScopes oldHub = Sentry.getCurrentScopes();
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java:78]: Rename to `oldScopes` and `clonedScopes`/`forkedScopes`?

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java [hunk 9]
```diff
@@ -87,20 +92,20 @@ public static <T> Flux<T> withSentry(final @NotNull Flux<T> flux) {
    */
   @ApiStatus.Experimental
   public static <T> Flux<T> withSentryNewMainHubClone(final @NotNull Flux<T> flux) {
-    final @NotNull IHub hub = Sentry.cloneMainHub();
+    final @NotNull IScopes hub = Sentry.cloneMainHub();
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java:95]: Rename to `scopes`

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java [hunk 10]
```diff
@@ -87,20 +92,20 @@ public static <T> Flux<T> withSentry(final @NotNull Flux<T> flux) {
...
+   * Writes the given Sentry {@link IScopes} to the {@link Context} and uses {@link
    * io.micrometer.context.ThreadLocalAccessor} to propagate it.
    *
    * <p>This requires - reactor.core.publisher.Hooks#enableAutomaticContextPropagation() to be
    * enabled - having `io.micrometer:context-propagation:1.0.2+` (provided by Spring Boot 3.0.3+) -
    * having `io.projectreactor:reactor-core:3.5.3+` (provided by Spring Boot 3.0.3+)
    */
   @ApiStatus.Experimental
-  public static <T> Flux<T> withSentryHub(final @NotNull Flux<T> flux, final @NotNull IHub hub) {
+  public static <T> Flux<T> withSentryHub(final @NotNull Flux<T> flux, final @NotNull IScopes hub) {
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java:108]: Rename to scopes

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryScheduleHook.java
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryScheduleHook.java:1]: Comment on class should refer to `scopes`

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryScheduleHook.java [hunk 11]
```diff
@@ -13,16 +13,18 @@
 @ApiStatus.Experimental
 public final class SentryScheduleHook implements Function<Runnable, Runnable> {
   @Override
+  @SuppressWarnings("deprecation")
   public Runnable apply(final @NotNull Runnable runnable) {
-    final IHub newHub = Sentry.getCurrentHub().clone();
+    // TODO fork instead
+    final IScopes newHub = Sentry.getCurrentScopes().clone();
```
Comment (lbloder) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryScheduleHook.java:19]: To be renamed, once forking is done

**Review by lbloder — COMMENTED** [review]
Comment on class `Reactive` in `SentryProperties` should refer to `scopes`.
Should we rename HubConfiguration in SentryAutoConfiguration?
Comment on class EnableSentry refer to Hub instead of Scopes.

Also, should we rename the SentryHubRegistrar and its methods?

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/SentryHubRegistrar.java
Comment (markushi) [inline sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/SentryHubRegistrar.java:1]: Should this be renamed to `SentryScopesRegistrar`?

**Review by markushi — APPROVED** [review]

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/AbstractSentryWebFilter.java (hunk 2 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/AbstractSentryWebFilter.java:47]: Changed in https://github.com/getsentry/sentry-java/pull/3366

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/AbstractSentryWebFilter.java (hunk 4 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/AbstractSentryWebFilter.java:79]: Changed in https://github.com/getsentry/sentry-java/pull/3366

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java (hunk 8 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java:78]: Changed in https://github.com/getsentry/sentry-java/pull/3366

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java (hunk 5 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java:26]: Changed in https://github.com/getsentry/sentry-java/pull/3366

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java (hunk 6 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java:42]: Changed in https://github.com/getsentry/sentry-java/pull/3366

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java (hunk 7 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java:55]: Changed in https://github.com/getsentry/sentry-java/pull/3366

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java (hunk 9 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java:95]: Changed in https://github.com/getsentry/sentry-java/pull/3366

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java (hunk 10 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/ReactorUtils.java:108]: Changed in https://github.com/getsentry/sentry-java/pull/3366

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryScheduleHook.java (hunk 11 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryScheduleHook.java:19]: Changed in https://github.com/getsentry/sentry-java/pull/3366

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryScheduleHook.java
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/SentryScheduleHook.java:1]: Changed in https://github.com/getsentry/sentry-java/pull/3366

File: sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/AbstractSentryWebFilter.java (hunk 3 above)
Comment (adinauer) [reply sentry-spring-jakarta/src/main/java/io/sentry/spring/jakarta/webflux/AbstractSentryWebFilter.java:66]: Changed in https://github.com/getsentry/sentry-java/pull/3366

**Comment by adinauer** [comment]
We need to test whether renaming Spring beans and config classes breaks anything. Even then user code could break by renaming them.

**Pull request merged**